# HP1 Project, YN Park

## 240716, HP1, ChIP-Seq

* HepG2, GM12878, K562 Encode Data
  * Input, Pooling per lane
  * Replicates -> Keep
* Motif Finding, MAC2 adj P value 0.05 and nolambda option

## 240705, Discussion

* HP1 ChIP-seq in ENCODE, (240703\_ChIP\_HP1\_ENCODE)
  * K562, GM12878, H1, HepG2 cell line
  * Peak patterns are similar to each other.
* From ENCODE Set,&#x20;
  * Some apoptosis-related genes overlap with other gene (ex, Bax or Bad)
  * Therefore, re-annotate the apoptotic gene set or target gene set, and perform annotate on random or non-target gene set as background data.
* HP1 ChIP-seq in Normal cell, (240628\_ChIP\_HP1\_GEO)
  * Genome Analyser II, The data is sub-optimal ..&#x20;

## 240517, Discussion

* PBS-NMDA: Hold
* changed to cell-line
  * GEO NGS, RNA-seq / H3K9me3 ChIP-seq/ ATAC-seq

## 240514, PBS-NMDA(Pax6set) Analysis

* RNA-seq basic profile
  * DEG Analysis (Heatmap, Volcano Plots ...)
  * Target Up-reg term Apoptosis
* (NEXT MEETING) Explain theoritcal background of RNA-seq and find the set of ATAC-seq
  * Next Meeting Schedule, 240516 after 6pm
